docs/root/start/quick-start/securing.rst

```diff
@@ -0,0 +1,198 @@
+
+.. _start_quick_start_securing_contexts:
+
+Upstream and downstream TLS contexts
+------------------------------------
+
+Machines connecting to Envoy to proxy traffic are "downstream" in relation to Envoy.
+
+Specifying a ``TLS`` context that clients can connect to is achieved by setting the
+:ref:`DownstreamTLSContext <envoy_v3_api_msg_extensions.transport_sockets.tls.v3.DownstreamTlsContext>`
+
+You will also need to provide valid certificates.
+
+.. literalinclude:: _include/envoy-demo-tls.yaml
+   :language: yaml
+   :linenos:
+   :lines: 1-37
+   :emphasize-lines: 28-37
+   :caption: :download:`envoy-demo-tls.yaml <_include/envoy-demo-tls.yaml>`
+
+Connecting to an "upstream" ``TLS`` service is conversely done by adding an
+:ref:`UpstreamTLSContext <envoy_v3_api_msg_extensions.transport_sockets.tls.v3.UpstreamTlsContext>`
+to the :ref:`cluster <envoy_v3_api_msg_config.cluster.v3.Cluster>`.
+
+.. literalinclude:: _include/envoy-demo-tls.yaml
+   :language: yaml
+   :linenos:
+   :lineno-start: 39
+   :lines: 39-57
+   :emphasize-lines: 16-19
+   :caption: :download:`envoy-demo-tls.yaml <_include/envoy-demo-tls.yaml>`
+
+.. _start_quick_start_securing_validation:
+
+Validate an endpoint's certificates when connecting
+---------------------------------------------------
+
+When Envoy connects to an upstream ``TLS`` service, it does not, by default, validate the certificates
+that it is presented with.
+
+You can use the :ref:`validation_context <envoy_v3_api_msg_extensions.transport_sockets.tls.v3.CertificateValidationContext>`
+to specify how Envoy should validate these certificates.
+
+Firstly, you can ensure that the certificates are from a mutually trusted certificate authority:
+
+.. literalinclude:: _include/envoy-demo-tls-validation.yaml
+   :language: yaml
+   :linenos:
+   :lineno-start: 43
+   :lines: 43-53
+   :emphasize-lines: 6-9
+   :caption: :download:`envoy-demo-tls-validation.yaml <_include/envoy-demo-tls-validation.yaml>`
+
+You can also ensure that the "Subject Alternative Names" for the cerficate match.
+
+This is commonly used by web certificates (X.509) to identify the domain or domains that a
+certificate is valid for.
+
+.. literalinclude:: _include/envoy-demo-tls-validation.yaml
+   :language: yaml
+   :linenos:
+   :lineno-start: 43
+   :lines: 43-53
+   :emphasize-lines: 6-7, 10-11
+   :caption: :download:`envoy-demo-tls-validation.yaml <_include/envoy-demo-tls-validation.yaml>`
+
+.. note::
+
+   If the "Subject Alternative Names" for a certificate are for a wildcard domain, eg ``*.example.com``,
+   this is what you should use when matching with ``match_subject_alt_names``.
+
+.. note::
+
+   See :ref:`here <envoy_v3_api_msg_extensions.transport_sockets.tls.v3.CertificateValidationContext>` to view all
+   of the possible configurations for certificate validation.
+
+.. _start_quick_start_securing_sni:
+
+Secure an endpoint with SNI
```

Comment:
Per above, SNI is about routing, not about security. It's actually not secure at all (until encrypted SNI in TLS 1.3). Can you check this around the doc and update/rephrase as needed?

Reply:
so i get its called routing but my understanding is that it just "addresses" the other side with an "sni" name

certainly in my testing, dns had no bearing here - as long as the sni name matched what was in the match list (or there was no  match list) then it worked

in terms of the title for this section - how about "Validating certificates with SNI" ?

i feel like we want to include sni here, and that it is sufficiently related to cert sec, but struggling to think how to word the section title